addons/mail/models/mail_thread.py

```diff
@@ -539,20 +577,14 @@ def message_change_thread(self, new_thread):
     # Automatic log / Tracking
     # ------------------------------------------------------
 
-    @api.model
+    @tools.ormcache()
     def _get_tracked_fields(self):
-        """ Return a structure of tracked fields for the current model.
-            :return dict: a dict mapping field name to description, containing on_change fields
-        """
-        tracked_fields = []
-        for name, field in self._fields.items():
-            tracking = getattr(field, 'tracking', None) or getattr(field, 'track_visibility', None)
-            if tracking:
-                tracked_fields.append(name)
-
-        if tracked_fields:
-            return self.fields_get(tracked_fields)
-        return {}
+        """ Return the set of tracked fields names for the current model. """
+        return {
+            name
+            for name, field in self._fields.items()
+            if getattr(field, 'tracking', None) or getattr(field, 'track_visibility', None)
```

Comment:
Shouldn't we finally deprecate/remove `track_visibility`?

Reply:
Yes, you can do that :wink: